orchestrator/test_runner/src/ibc_metadata.rs

```diff
@@ -0,0 +1,188 @@
+//! This is a test for the ibc metadata proposal, which is a way to set the denom metadata for a given
+
+use crate::airdrop_proposal::wait_for_proposals_to_execute;
+use crate::happy_path_v2::deploy_cosmos_representing_erc20_and_check_adoption;
+use crate::utils::{vote_yes_on_proposals, ValidatorKeys};
+use crate::{get_deposit, get_fee, TOTAL_TIMEOUT};
+use clarity::Address;
+use deep_space::utils::encode_any;
+use deep_space::Contact;
+use gravity_proto::cosmos_sdk_proto::cosmos::bank::v1beta1::{DenomUnit, Metadata};
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use gravity_proto::gravity::IbcMetadataProposal;
+use tonic::transport::Channel;
+use web30::client::Web3;
+
+pub async fn ibc_metadata_proposal_test(
+    gravity_address: Address,
+    keys: Vec<ValidatorKeys>,
+    grpc_client: GravityQueryClient<Channel>,
+    contact: &Contact,
+    web30: &Web3,
+) {
+    let mut grpc_client = grpc_client;
+    let metadata_start = contact.get_all_denoms_metadata().await.unwrap();
+    info!("Starting metadata {:?}", metadata_start);
+    // the token we want to try and set the metadata of
+    let target_denom = "ibc/nometadatatoken".to_string();
+    // already has metadta set, should fail
+    let invalid_denom = "footoken".to_string();
+
+    // totally invalid metadata to test
+    let invalid_metadata = Metadata {
+        description: "invalid".to_string(),
+        denom_units: Vec::new(),
+        base: "no good".to_string(),
+        display: "nada".to_string(),
+        name: "invalid".to_string(),
+        symbol: "INV".to_string(),
+    };
+
+    // valid metadata to test, but would overwrite existing metadata
+    let foo_metadata = Metadata {
+        description: "A testing token".to_string(),
+        denom_units: vec![
+            DenomUnit {
+                denom: "footoken".to_string(),
+                exponent: 0,
+                aliases: vec![],
+            },
+            DenomUnit {
+                denom: "mfootoken".to_string(),
+                exponent: 6,
+                aliases: vec![],
+            },
+        ],
+        base: "footoken".to_string(),
+        display: "mfootoken".to_string(),
+        name: "Footoken".to_string(),
+        symbol: "FOO".to_string(),
+    };
+
+    // valid metadata to test, but would overwrite existing metadata
```

Comment:
Update comment

Reply:
Thanks for flagging added another case that does try that.